File: MobileApp/app/src/main/java/com/elearning/ui/data/model/Enrollment.kt
package com.elearning.ui.data.model

import com.google.gson.annotations.SerializedName

data class Enrollment(
    val id: Int,
    @SerializedName("student_id")
    val studentId: Int,
    @SerializedName("course_id")
    val courseId: Int,
    @SerializedName("enrolled_at")
    val enrolledAt: String,
    val progress: Float = 0f,
    
    // Detailed fields from EnrollmentDetailResponse
    @SerializedName("course_title")
    val courseTitle: String? = null,
    @SerializedName("course_description")
    val courseDescription: String? = null,
    @SerializedName("teacher_username")
    val teacherName: String? = null
)

File: Backend/app/schemas/enrollment.py
from datetime import datetime
from pydantic import BaseModel


class EnrollmentBase(BaseModel):
    """Base enrollment schema."""
    course_id: int


class EnrollmentCreate(EnrollmentBase):
    """Enrollment creation schema."""
    pass


class EnrollmentResponse(BaseModel):
    """Enrollment response schema."""
    id: int
    student_id: int
    course_id: int
    enrolled_at: datetime
    
    class Config:
        from_attributes = True


class EnrollmentDetailResponse(EnrollmentResponse):
    """Detailed enrollment response with course info."""
    course_title: str | None = None
    course_description: str | None = None
    teacher_username: str | None = None
